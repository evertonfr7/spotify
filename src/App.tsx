import { Routes, Route, BrowserRouter as Router } from 'react-router-dom'
import Home from './routes/Home'
import Artist from './routes/Artist'
import Artists from './routes/Artists'
import Login from './routes/Login'
import Profile from './routes/Profile'
import Playlists from './routes/Playlists'
import ProtectedRoutes from './utils/ProtectedRoutes'
import { useEffect } from 'react'
import { getTokenFromUrl } from './utils/getTokenFromUrl'

import Callback from './routes/Callback'
import Cookies from 'js-cookie'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

const queryClient = new QueryClient()

const App = () => {
  const savedToken = Cookies.get('spotify_token')
  useEffect(() => {
    const _spotifyToken = getTokenFromUrl().access_token
    window.location.hash = ''

    if (_spotifyToken) {
      Cookies.set('spotify_token', _spotifyToken, {
        expires: new Date(Date.now() + 3600 * 1000),
      })
    }
  }, [savedToken])

  return (
    <QueryClientProvider client={queryClient}>
      <Router>
        <Routes>
          <Route path="/login" element={<Login />} />
          <Route path="/callback" element={<Callback />} />

          <Route element={<ProtectedRoutes />}>
            <Route path="/" element={<Home />} />
            <Route path="/artistas" element={<Artists />} />
            <Route path="/artistas/:id" element={<Artist />} />
            <Route path="/playlists" element={<Playlists />} />
            <Route path="/perfil" element={<Profile />} />
          </Route>
        </Routes>
      </Router>
    </QueryClientProvider>
  )
}

export default App
