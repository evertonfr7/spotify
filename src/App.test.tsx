import { render, screen } from '@testing-library/react'
import App from './App'

// Test suite for the App component
describe('App Component', () => {
  test('renders the Button component with correct text', () => {
    render(<App />)

    const buttonElement = screen.getByText(/Ol√° Mundo/i)
    expect(buttonElement).toBeInTheDocument()
  })
})
