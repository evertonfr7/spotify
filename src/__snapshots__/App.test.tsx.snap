// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders the Button component with correct text 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-screen flex flex-col items-center justify-center gap-4"
      >
        <div>
          <img
            alt="Spotify Logo"
            aria-label="Spotify Logo"
            class="w-[114.8px] h-[35px] md:w-[164px] md:h-[50px] block"
            data-testid="logo"
            height="50"
            role="img"
            src="[object Object]"
            title="Spotify"
            width="164"
          />
        </div>
        <p
          class="font-default text-[12px]/[18px] font-medium text-white md:text-[14px]/[20px]"
        >
          Entra com sua conta Spotify clicando no botão abaixo
        </p>
        <a
          href="https://accounts.spotify.com/authorize?client_id=undefined&redirect_uri=undefined&scope=user-read-email user-read-private user-read-playback-state user-modify-playback-state user-read-currently-playing user-read-playback-position user-read-recently-played user-top-read user-library-read user-library-modify playlist-read-private playlist-read-collaborative playlist-modify-private playlist-modify-public streaming user-follow-read user-follow-modify&response_type=token&show_dialog=true"
        >
          <button
            class="bg-[#57B660] text-[#181414] rounded-3xl px-8 h-[32px] font-default text-[14px]/[18px] font-bold md:h-[42px] md:px-10 md:text-base/[20px] focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-[#57B660]"
            type="button"
          >
            Entrar
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-screen flex flex-col items-center justify-center gap-4"
    >
      <div>
        <img
          alt="Spotify Logo"
          aria-label="Spotify Logo"
          class="w-[114.8px] h-[35px] md:w-[164px] md:h-[50px] block"
          data-testid="logo"
          height="50"
          role="img"
          src="[object Object]"
          title="Spotify"
          width="164"
        />
      </div>
      <p
        class="font-default text-[12px]/[18px] font-medium text-white md:text-[14px]/[20px]"
      >
        Entra com sua conta Spotify clicando no botão abaixo
      </p>
      <a
        href="https://accounts.spotify.com/authorize?client_id=undefined&redirect_uri=undefined&scope=user-read-email user-read-private user-read-playback-state user-modify-playback-state user-read-currently-playing user-read-playback-position user-read-recently-played user-top-read user-library-read user-library-modify playlist-read-private playlist-read-collaborative playlist-modify-private playlist-modify-public streaming user-follow-read user-follow-modify&response_type=token&show_dialog=true"
      >
        <button
          class="bg-[#57B660] text-[#181414] rounded-3xl px-8 h-[32px] font-default text-[14px]/[18px] font-bold md:h-[42px] md:px-10 md:text-base/[20px] focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-[#57B660]"
          type="button"
        >
          Entrar
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
